### PR TITLE
feat(Core/Battleground): rework bg queue system

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- Port battleground queue system from TC
- Delete code for reenter queue (skip spam in world chat)
- Add more check before enter queue
- Add more logs `bg.battleground`

## Issues Addressed:
- Maybe close https://github.com/chromiecraft/chromiecraft/issues/335

## SOURCE:
- TC

## Tests Performed:
- Tested all bg queues
- Tested arena skirmish queue 2v2
- Tested arena rated queue 2v2
- Tested group queue continion `IsPlayerInvitedToRatedArena`

## How to Test the Changes:
1. Test random/normal bg queue without `.debug bg`
2. Test rated/skirmish arena queue without `.debug arena`
3. Test `mod-cfbg` with PR https://github.com/azerothcore/mod-cfbg/pull/97 (random/normal bg queue)
4. Test `mod-queue-list-cache` with PR https://github.com/azerothcore/mod-queue-list-cache/pull/5
5. Test `mod-low-level-arena` with PR https://github.com/azerothcore/mod-low-level-arena/pull/7

## Modules dependence:
- [ ] https://github.com/azerothcore/mod-cfbg/pull/97
- [ ] https://github.com/azerothcore/mod-queue-list-cache/pull/5
- [ ] https://github.com/azerothcore/mod-low-level-arena/pull/7
- [ ] https://github.com/azerothcore/mod-low-level-rbg/pull/2

## Known Issues and TODO List:
- [x] Rework hooks
- [x] Rework cfbg

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by temperrr** [comment]
When queing random bg as level 80: it popped and then we all joined the bg and then it crashed the server:
Rest worked fine
```

Thread 1 "worldserver" received signal SIGSEGV, Segmentation fault.
0x000000000224b5ef in Acore::Abort (file="/root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp", line=273, function="RemovePlayer", fmtMessage="BattlegroundQueue: couldn't find player Restososos (GUID Full: 0x", '0' <repeats 13 times>, "9a7 Type: Player Low: 2471)") at /root/env/azerothcore-wotlk/src/common/Debugging/Errors.cpp:145
145	    Crash(formattedMessage.c_str());
#0  0x000000000224b5ef in Acore::Abort (file="/root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp", line=273, function="RemovePlayer", fmtMessage="BattlegroundQueue: couldn't find player Restososos (GUID Full: 0x", '0' <repeats 13 times>, "9a7 Type: Player Low: 2471)") at /root/env/azerothcore-wotlk/src/common/Debugging/Errors.cpp:145
#1  0x0000000001165885 in Acore::Abort<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > (file="/root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp", line=273, function="RemovePlayer", fmt="BattlegroundQueue: couldn't find player {} ({})", args=..., args=...) at /root/env/azerothcore-wotlk/src/common/Debugging/Errors.h:46
#2  0x000000000115c773 in BattlegroundQueue::RemovePlayer (this=0x3027210 <BattlegroundMgr::instance()::instance+38464>, guid=..., decreaseInvitedCount=true) at /root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp:273
#3  0x0000000001b55117 in WorldSession::LogoutPlayer (this=0x7fd2e5743800, save=true) at /root/env/azerothcore-wotlk/src/server/game/Server/WorldSession.cpp:591
#4  0x0000000001863d9b in WorldSession::HandleLogoutRequestOpcode (this=0x7fd2e5743800) at /root/env/azerothcore-wotlk/src/server/game/Handlers/MiscHandler.cpp:452
#5  0x0000000001d3e7cb in PacketHandler<WorldPackets::Character::LogoutRequest, &WorldSession::HandleLogoutRequestOpcode>::Call (this=0x7fd31054f920, session=0x7fd2e5743800, packet=...) at /root/env/azerothcore-wotlk/src/server/game/Server/Protocol/Opcodes.cpp:35
#6  0x0000000001b584d1 in WorldSession::Update (this=0x7fd2e5743800, diff=11, updater=...) at /root/env/azerothcore-wotlk/src/server/game/Server/WorldSession.cpp:349
#7  0x0000000001f7018f in World::UpdateSessions (this=0x7fd34d1bd000, diff=11) at /root/env/azerothcore-wotlk/src/server/game/World/World.cpp:2905
#8  0x0000000001f68a78 in World::Update (this=0x7fd34d1bd000, diff=11) at /root/env/azerothcore-wotlk/src/server/game/World/World.cpp:2327
#9  0x00000000004902ba in WorldUpdateLoop () at /root/env/azerothcore-wotlk/src/server/worldserver/Main.cpp:588
#10 0x000000000048a7f7 in main (argc=3, argv=0x7ffd40e2a158) at /root/env/azerothcore-wotlk/src/server/worldserver/Main.cpp:419
#0  0x000000000224b5ef in Acore::Abort (file="/root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp", line=273, function="RemovePlayer", fmtMessage="BattlegroundQueue: couldn't find player Restososos (GUID Full: 0x", '0' <repeats 13 times>, "9a7 Type: Player Low: 2471)") at /root/env/azerothcore-wotlk/src/common/Debugging/Errors.cpp:145
        formattedMessage = "\n#", '-' <repeats 70 times>, "#\n \n>> ABORTED\n\n# Location '/root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp:273'\n# Function 'Rem"...
#1  0x0000000001165885 in Acore::Abort<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > > (file="/root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp", line=273, function="RemovePlayer", fmt="BattlegroundQueue: couldn't find player {} ({})", args=..., args=...) at /root/env/azerothcore-wotlk/src/common/Debugging/Errors.h:46
No locals.
#2  0x000000000115c773 in BattlegroundQueue::RemovePlayer (this=0x3027210 <BattlegroundMgr::instance()::instance+38464>, guid=..., decreaseInvitedCount=true) at /root/env/azerothcore-wotlk/src/server/game/Battlegrounds/BattlegroundQueue.cpp:273
        playerName = "Restososos"
        itr = {first = {static Empty = {static Empty = <same as static member of an already seen type>, _guid = 0}, _guid = 0}, second = 0x3027240 <BattlegroundMgr::instance()::instance+38512>}
        groupInfo = 0x30412d8 <ScriptMgr::instance()::instance>
        _bracketId = 32722
        _groupType = 4260979712
        group_itr = 0x1
        pitr = {static Empty = {static Empty = <same as static member of an already seen type>, _guid = 0}, _guid = 21235104}
#3  0x0000000001b55117 in WorldSession::LogoutPlayer (this=0x7fd2e5743800, save=true) at /root/env/azerothcore-wotlk/src/server/game/Server/WorldSession.cpp:591
        bgQueueTypeId = BATTLEGROUND_QUEUE_RB
        i = 0
        stmt = 0x7fd2e4e050b0
#4  0x0000000001863d9b in WorldSession::HandleLogoutRequestOpcode (this=0x7fd2e5743800) at /root/env/azerothcore-wotlk/src/server/game/Handlers/MiscHandler.cpp:452
        instantLogout = true
        preventAfkSanctuaryLogout = false
        preventAfkLogout = false
        canLogoutInCombat = true
        reason = 0
        logoutResponse = {<WorldPackets::ServerPacket> = {<WorldPackets::Packet> = {_vptr$Packet = 0x26432a0 <vtable for WorldPackets::Character::LogoutResponse+16>, _worldPacket = {<ByteBuffer> = {_vptr$ByteBuffer = 0x23fa838 <vtable for WorldPacket+16>, static DEFAULT_SIZE = 4096, _rpos = 0, _wpos = 5, _storage = std::vector of length 5, capacity 5 = {0 '\000', 0 '\000', 0 '\000', 0 '\000', 1 '\001'}}, m_opcode = 76, m_receivedTime = {__d = {__r = 0}}}}, <No data fields>}, LogoutResult = 0, Instant = true}
#5  0x0000000001d3e7cb in PacketHandler<WorldPackets::Character::LogoutRequest, &WorldSession::HandleLogoutRequestOpcode>::Call (this=0x7fd31054f920, session=0x7fd2e5743800, packet=...) at /root/env/azerothcore-wotlk/src/server/game/Server/Protocol/Opcodes.cpp:35
        nicePacket = {<WorldPackets::ClientPacket> = {<WorldPackets::Packet> = {_vptr$Packet = 0x264c2c8 <vtable for WorldPackets::Character::LogoutRequest+16>, _worldPacket = {<ByteBuffer> = {_vptr$ByteBuffer = 0x23fa838 <vtable for WorldPacket+16>, static DEFAULT_SIZE = 4096, _rpos = 0, _wpos = 0, _storage = std::vector of length 0, capacity 0}, m_opcode = 75, m_receivedTime = {__d = {__r = 0}}}}, <No data fields>}, <No data fields>}
#6  0x0000000001b584d1 in WorldSession::Update (this=0x7fd2e5743800, diff=11, updater=...) at /root/env/azerothcore-wotlk/src/server/game/Server/WorldSession.cpp:349
        opcode = CMSG_LOGOUT_REQUEST
        opHandle = 0x7fd31054f920
        ihe = @0x2351e65: {<WorldPackets::InvalidStringValueException> = {<ByteBufferInvalidValueException> = {<ByteBufferException> = {<std::exception> = {<No data fields>}, msg_ = <error: Cannot access memory at address 0x48e5894855000044>}, <No data fields>}, _value = <error: Cannot access memory at address 0x48e5894855000044>}, <No data fields>}
        ihe = @0x7ffd40e27b58: {<WorldPackets::InvalidStringValueException> = {<ByteBufferInvalidValueException> = {<ByteBufferException> = {<std::exception> = {<No data fields>}, msg_ = <error: Cannot access memory at address 0x9d426>}, <No data fields>}, _value = "`\271\003\003\000\000\000\000\001", '\000' <repeats 25 times>, "\200?", '\000' <repeats 20 times>, "\230\271\003\003\000\000\000\000\001", '\000' <repeats 25 times>, "\200?", '\000' <repeats 20 times>, "й\003\003\000\000\000\000\001", '\000' <repeats 25 times>, "\200?", '\000' <repeats 20 times>, "\b\272\003\003\000\000\000\000\001", '\000' <repeats 23 times>...}, <No data fields>}
        pamce = @0x7fd35b934760: {<ByteBufferException> = {<std::exception> = {<No data fields>}, msg_ = ""}, <No data fields>}
        packet = 0x7fd2cf5a71c0
        deletePacket = true
        requeuePackets = std::vector of length 0, capacity 0
        processedPackets = 0
        currentTime = 1646381165
#7  0x0000000001f7018f in World::UpdateSessions (this=0x7fd34d1bd000, diff=11) at /root/env/azerothcore-wotlk/src/server/game/World/World.cpp:2905
        pSession = 0x7fd2e5743800
        updater = {<PacketFilter> = {_vptr$PacketFilter = 0x2640d90 <vtable for WorldSessionFilter+16>, m_pSession = 0x7fd2e5743800}, <No data fields>}
        currentSessionId = 48677
        itr = {<std::__detail::_Node_iterator_base<std::pair<unsigned int const, WorldSession*>, false>> = {_M_cur = 0x7fd2fc9a1520}, <No data fields>}
        next = {<std::__detail::_Node_iterator_base<std::pair<unsigned int const, WorldSession*>, false>> = {_M_cur = 0x7fd2fc987860}, <No data fields>}
        currTime = 32723
#8  0x0000000001f68a78 in World::Update (this=0x7fd34d1bd000, diff=11) at /root/env/azerothcore-wotlk/src/server/game/World/World.cpp:2327
        __ac_metric_stop_watch2326 = {_logger = {<No data fields>}, _startTime = {__d = {__r = 0}}}
        guard = {_M_device = @0x3040588}
        __ac_metric_stop_watch2218 = {_logger = {<No data fields>}, _startTime = {__d = {__r = 0}}}
        currentGameTime = {__r = 1646381165}
#9  0x00000000004902ba in WorldUpdateLoop () at /root/env/azerothcore-wotlk/src/server/worldserver/Main.cpp:588
        diff = 11
        executionTimeDiff = 0
        realCurrTime = 644145
        realPrevTime = 644134
#10 0x000000000048a7f7 in main (argc=3, argv=0x7ffd40e2a158) at /root/env/azerothcore-wotlk/src/server/worldserver/Main.cpp:419
        configFile = "/root/env/azerothcore-wotlk/env/dist/etc/worldserver.conf"
        c = 3
        ioContext = warning: 4322860.793230 RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<Acore::Asio::IoContext, std::allocator<Acore::Asio::IoContext>, (__gnu_cxx::_Lock_policy)2>'
4322860.793236 
warning: 4322860.793335 RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<Acore::Asio::IoContext, std::allocator<Acore::Asio::IoContext>, (__gnu_cxx::_Lock_policy)2>'
4322860.793339 
std::shared_ptr<Acore::Asio::IoContext> (use count 4, weak count 0) = {get() = 0x7fd358addd30}
        opensslHandle = warning: 4322860.975300 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_2, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322860.975305 
warning: 4322860.975411 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_2, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322860.975414 
std::shared_ptr<void> (use count 1, weak count 0) = {get() = 0x0}
        seed = {_bn = 0x7fd358ae8f80}
        pidFile = "./world.pid"
        signals = {impl_ = {service_ = 0x7fd358a76500, implementation_ = {queue_ = {<boost::asio::detail::noncopyable> = {<No data fields>}, front_ = 0x7fd358a25ab0, back_ = 0x7fd358a25ab0}, signals_ = 0x7fd358a1e810}, implementation_executor_ = {executor_ = {impl_ = 0x7fd358a1e570}, has_native_impl_ = false}}}
        numThreads = 2
        threadPool = warning: 4322861.252308 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::vector<std::thread, std::allocator<std::thread> >*, main::$_3, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.252314 
warning: 4322861.252407 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::vector<std::thread, std::allocator<std::thread> >*, main::$_3, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.252411 
std::shared_ptr<std::vector<std::thread, std::allocator<std::thread> >> (use count 1, weak count 0) = {get() = 0x7fd358add0a0}
        sScriptMgrHandle = warning: 4322861.433485 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_5, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.433491 
warning: 4322861.433590 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_5, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.433593 
std::shared_ptr<void> (use count 1, weak count 0) = {get() = 0x0}
        dbHandle = warning: 4322861.602303 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_6, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.602308 
warning: 4322861.602412 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_6, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.602415 
std::shared_ptr<void> (use count 1, weak count 0) = {get() = 0x0}
        sMetricHandle = warning: 4322861.769896 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_8, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.769902 
warning: 4322861.770010 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_8, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.770013 
std::shared_ptr<void> (use count 1, weak count 0) = {get() = 0x0}
        mapManagementHandle = warning: 4322861.934971 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_9, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.934977 
warning: 4322861.935068 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_9, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322861.935071 
std::shared_ptr<void> (use count 1, weak count 0) = {get() = 0x0}
        raAcceptor = std::unique_ptr<AsyncAcceptor> = {get() = 0x0}
        soapThread = warning: 4322862.133129 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::thread*, main::$_10, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.133135 
warning: 4322862.133250 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::thread*, main::$_10, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.133253 
std::shared_ptr<std::thread> (use count 1, weak count 0) = {get() = 0x7fd2fb43d2a0}
        worldPort = 40000
        worldListener = "0.0.0.0"
        networkThreads = 2
        sWorldSocketMgrHandle = warning: 4322862.322268 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_11, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.322274 
warning: 4322862.322395 RTTI symbol not found for class 'std::_Sp_counted_deleter<decltype(nullptr), main::$_11, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.322399 
std::shared_ptr<void> (use count 1, weak count 0) = {get() = 0x0}
        freezeDetector = warning: 4322862.509530 RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<FreezeDetector, std::allocator<FreezeDetector>, (__gnu_cxx::_Lock_policy)2>'
4322862.509536 
warning: 4322862.509642 RTTI symbol not found for class 'std::_Sp_counted_ptr_inplace<FreezeDetector, std::allocator<FreezeDetector>, (__gnu_cxx::_Lock_policy)2>'
4322862.509645 
std::shared_ptr<FreezeDetector> (use count 1, weak count 1) = {get() = 0x7fd34d1afba0}
        cliThread = warning: 4322862.713083 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::thread*, void (*)(std::thread*), std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.713089 
warning: 4322862.713207 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::thread*, void (*)(std::thread*), std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.713210 
std::shared_ptr<std::thread> (use count 1, weak count 0) = {get() = 0x7fd2fb43d2b8}
        auctionLisingThread = warning: 4322862.904632 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::thread*, main::$_12, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.904638 
warning: 4322862.904753 RTTI symbol not found for class 'std::_Sp_counted_deleter<std::thread*, main::$_12, std::allocator<void>, (__gnu_cxx::_Lock_policy)2>'
4322862.904757 
std::shared_ptr<std::thread> (use count 1, weak count 0) = {get() = 0x7fd2fb43d2c0}
```

**Comment by Winfidonarleyan** [comment]
@temperrr try again

**Comment by IntelligentQuantum** [comment]
@temperrr If you are testing this PR, please try this Arena as well (Blades's Edge Arena (arena) Thank you

**Comment by temperrr** [comment]
We tested skirmish arena at 80 2v2 and 3v3. Didn't get blade edge but others

**Comment by Winfidonarleyan** [comment]
> We tested skirmish arena at 80 2v2 and 3v3. Didn't get blade edge but others

So, are there any problems that need to be solved?

**Comment by temperrr** [comment]
I did not test after the crash fix.

**Comment by temperrr** [comment]
Tested with crash fix, no crash anymore seems to work fine as far as we could test.

**Comment by IntelligentQuantum** [comment]
So The problem of not joining this arena is not related to this PR?

**Comment by temperrr** [comment]
Oh didnt check blade edges can do

**Comment by temperrr** [comment]
2v2 queue didnt pop on arena now :? (As lvl 70)

**Comment by temperrr** [comment]
image

**Comment by Winfidonarleyan** [comment]
Test with `LLA` and level 80
image

Start test low level...

**Comment by temperrr** [comment]
I tested without debug arena

**Comment by Winfidonarleyan** [comment]
> I tested without debug arena

can retest?

File: src/server/game/Battlegrounds/BattlegroundQueue.cpp [hunk 1]
```diff
@@ -377,504 +389,562 @@ bool BattlegroundQueue::GetPlayerGroupInfoData(ObjectGuid guid, GroupQueueInfo*
...
-    int32 hordeDiff = hordeFree - int32(m_SelectionPools[TEAM_HORDE].GetPlayerCount());
+                aliFree = std::max(aliFree, 0);
+                hordeFree = std::max(hordeFree, 0);
+            }
+        }
+    }
 
-    int32 invType = sWorld->getIntConfig(CONFIG_BATTLEGROUND_INVITATION_TYPE);
-    int32 invDiff = 0;
+    //iterator for iterating through bg queue
```
Comment (Nefertumm) [inline src/server/game/Battlegrounds/BattlegroundQueue.cpp:429]: Redundant, iterator for iterating... 5Head

**Review by Nefertumm — APPROVED** [review]

**Comment by Nefertumm** [comment]
Please, add changelogs for the changes on BGScript. Thank you!

File: src/server/game/Battlegrounds/BattlegroundMgr.cpp [hunk 2]
```diff
@@ -636,11 +746,26 @@ void BattlegroundMgr::SetHolidayWeekends(uint32 mask)
...
+    if (std::find(m_QueueUpdateScheduler.begin(), m_QueueUpdateScheduler.end(), scheduleId) == m_QueueUpdateScheduler.end())
+        m_QueueUpdateScheduler.emplace_back(scheduleId);
+}
+
+uint32 BattlegroundMgr::GetMaxRatingDifference() const
 {
-    uint64 const scheduleId = ((uint64)arenaRatedTeamId << 32) | (bgQueueTypeId << 16) | bracket_id;
-    if (std::find(m_ArenaQueueUpdateScheduler.begin(), m_ArenaQueueUpdateScheduler.end(), scheduleId) == m_ArenaQueueUpdateScheduler.end())
-        m_ArenaQueueUpdateScheduler.push_back(scheduleId);
+    // this is for stupid people who can't use brain and set max rating difference to 0
```
[the remark this answers is not in the document]
Comment (Winfidonarleyan) [reply src/server/game/Battlegrounds/BattlegroundMgr.cpp:760]: my bad, sorry

**Pull request merged**